spec/sidekiq_poison_pill_remedy_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe SidekiqPoisonPillRemedy do
+  describe ".remedy" do
+    subject(:call) { described_class.remedy.call(nil, job) }
+
+    let(:default_queue) { "default" }
+    let(:poison_pill_queue) { "poison_pill" }
+    let(:enqueue_job) { MyJob.set(queue: job_queue).perform_async("fail") }
+    let(:job) { Sidekiq::Queue.new(job_queue).find_job(enqueue_job) }
+
+    before do
+      Sidekiq::Testing.disable!
+      Sidekiq::Queue.new(poison_pill_queue).clear
+      Sidekiq::Queue.new(default_queue).clear
+
+      enqueue_job
+
+      allow_any_instance_of(Sidekiq::DeadSet).to receive(:find_job).with(enqueue_job).and_return(job)
+      allow(Sentry).to receive(:capture_message).and_call_original
+      allow(Sidekiq.logger).to receive(:fatal)
```

Comment:
@Robgra13 please always use `and_call_original` if you use it for the purpose of spying (i.e. using have_received later)